apps/timetable/ui/js/index.js

```diff
@@ -70,6 +59,31 @@ define(['gh.core', 'gh.constants', 'validator', 'gh.calendar', 'gh.header', 'gh.
         });
     };
 
+    /**
+     * Show the empty timetable notification
+     *
+     * @param  {Object}    evt     The dispatched jQuery event
+     * @param  {Object}    data    Object containing the module data
+     * @private
+     */
+    var showEmptyTimetable = function(evt, data) {
+        // Render the 'empty-timetable' template
```

Comment:
There are 2 different formats coming in for data, you can verify this by adding an external link to a part and executing this function by: adjusting the part pickers to select the part with an external link or refreshing the page. I got this fixed by adding `data = data.record || data;` to the beginning of the file